packages/context/src/binding.ts

```diff
@@ -16,21 +16,55 @@ export type ValueOrPromise<T> = T | Promise<T>;
 // FIXME(bajtos) The binding class should be parameterized by the value
 // type stored
 export class Binding {
-  private _tagName: string;
+  static PROPERTY_SEPARATOR = '/';
+  static KEY_SEPARATOR = '.';
+  static KEY_SEPARATOR_SLASH = ':';
+  /**
+   * Validate the binding key format. Please note that `.` is reserved.
+   * @param key Binding key
+   */
+  static validateKey(key: string) {
+    if (!key) throw new Error('Binding key must be provided.');
+    if (key.indexOf(Binding.PROPERTY_SEPARATOR) !== -1) {
+      throw new Error(`Binding key ${key} cannot contain`
+        + ` '${Binding.PROPERTY_SEPARATOR}'. Please use`
+        + ` '${Binding.KEY_SEPARATOR}' or`
+        + ` '${Binding.KEY_SEPARATOR_SLASH}' instead.`);
+    }
+    return key;
+  }
+
+  static normalizeKey(key: string) {
```

Comment:
👍, but all the other code in the codebase declares them explicitly. Anyways, we can start removing all the other ones too.